### DISCUSSION TITLE
Reload a browser page on C-c

### DISCUSSION THREAD

**Discussion by aykaramba** [rabbibotton/clog#329] [General]
When I am working in CLOG, I spend a good amount of time compiling functions with C-c and going back and forth between the repl, file and manually reloading the browser page  to test ideas out.  This works great and has helped me learn a lot.

I am finding that as I learn CLOG I can write enough chunks of code that it would be nice to be able to Do something like C-c C-v (unmapped in both Emacs and LEM) and have the editor compile the form and then reload the current browser url.

So, the editor function would basically do the equivalent of C-c C-c and then call a function such as (url-replace (location body) "here").

This would reduce the amount of time I need to switch windows and use the mouse or remove the need for the browser to reload every x seconds using a plugin.

Is there a way to to query CLOG for the current url and then use something like (url-replace (location body) "here") ?  I tried going into (in-package :mypackage) and trying (url-replace (location body) "/") but the variable BODY is unbound even though :mypackage is currently running.  

Suggestions?

**Comment by aykaramba** [comment]
I figured out that I can place my functions in the standard "on-new-window (body) ..." function and access it directly.  I just have to write a LEM command to handle the compile / reload calls.

This function reloads the page:

```
(defun reload (obj)
   (url-replace (location body) "/"))
```

**Comment by aykaramba** [comment]
EDITOR REQUIREMENTS: LEM or Emacs

Just wanted to follow up in case some user bumps into this thread.   Here is how you can create a shortcut to compile a form and do some extra things at the same time.  In my case:

1) I am using the LEM editor.

2) Want to be able to compile forms and reload the CLOG web page so I put the following function into the main (defun on-new-window () ...) form as follows:

```
(defun reload (obj)
  (url-replace (location body) "/"))
```

3) We can now create a new key binding to be able to compile a form and then call the (reload ()) function to reload the CLOG page:

```
(in-package :lem-user)
(ql:quickload :my-app)
(define-key lem-lisp-mode:*lisp-mode-keymap* "C-c C-v" 'clog-reload)
(define-command clog-reload () ()
      (my-app::reload 123)
  )
``` 

4) For me this reduces the number of round trips to reload the page each time I try something new.  It also opens up an insight into how we can start to use LEM and get additional interactivity while developing in CLOG.   

Thanks for the help everyone!

CLOG + Lisp are pure awesome.

**Comment by rabbibotton** [comment]
Take a look at the new additions to tutorial 22 I just pushed on popup browser windows and tabs, you may find it a good alternative

**Comment by aykaramba** [comment]
Thank you for the heads up, will look into it tonight.
